test/jdk/java/awt/im/4959409/bug4959409.java

```diff
@@ -22,51 +22,173 @@
  */
 
 /**
- *
+ * @test
  * @bug 4959409
- * @author Naoto Sato
+ * @summary Check whether pressing SHIFT + 1 triggers key event
+ * @key headful
  */
 
-import java.awt.*;
-import java.awt.event.*;
-import javax.swing.*;
+import javax.swing.JFrame;
+import javax.swing.JTextField;
+import javax.swing.JLabel;
+import javax.swing.SwingUtilities;
+import java.awt.event.KeyAdapter;
+import java.awt.event.ComponentAdapter;
+import java.awt.event.ComponentEvent;
+import java.awt.event.KeyEvent;
+import java.awt.event.InputEvent;
+
+import java.awt.BorderLayout;
+import java.awt.Container;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class bug4959409 {
+
+    public final static int TIMEOUT = 30;
+    public final static int DELAY = 300;
+
+    public JFrame createUIAndTest() throws Exception {
+        final JFrame frame = new JFrame("Test bug4959409");
```

Comment:
Please make sure that the Swing components, created/used/disposed on the EDT.

Reply:
Added EDT